### ISSUE TITLE
Error in function login: Forbidden

### ISSUE THREAD

**Issue by tmwebb2** [tgorgdotcom/locast2plex#272]
I ma getting the error below. I allowed a Microsoft update and rebooted the laptop. Happens on my desktop and laptop.

C:\locast2plex>python3 main.py
Initiating Locast2Plex v0.6.5
Opening and Verifying Configuration File.
C:\locast2plex\config\config.ini
C:\locast2plex\config.ini
Loading Configuration File: C:\locast2plex\config.ini
Tuner count set to 3
Server is set to run on  127.0.0.1:6077
UUID set to: zkqzpils...
Getting user location...
Getting location via IP Address.
Got external IP 99.64.87.111.
Got location as Atlanta - DMA 524 - Lat\Lon 34.0663\-84.6784
Logging into Locast using username notmyreal@comcast.net...
Error in function login: Forbidden
Invalid Locast Login Credentials. Exiting...

**Comment by rqbanerjee** [comment]
I am also seeing this, starting today. docker on a Synology NAS (Linux). I can use the same credentials to log into the website. Maybe Locast did a bad deployment, or changed their API endpoint for logins?

**Comment by jg-24** [comment]
Locast apparently started enforcing a captcha implemented with Google ReCaptcha.

**Comment by seakurt** [comment]
Same issue after restarting. Windows 10.

**Comment by jg-24** [comment]
For the developers: try using a hardcoded client ID and captcha in the locast api calls.  The ID is currently hardcoded in website static objects so it may change and the captcha appears have just been added as a required field but is not being enforced/checked yet (so it will certainly change). Kudos to other developers for figuring this out so quickly...

https://api.locastnet.org/api/user/login?client_id=CqhAMsBw%2BnxTXSJMLGqyOw%3D%3D

credentials = {
        "username": username,
        "password": password,
        "captcha": "anything"
    }

**Comment by purduepete** [comment]
> For the developers: try using a hardcoded client ID and captcha in the locast api calls. The ID is currently hardcoded in website static objects so it may change and the captcha appears have just been added as a required field but is not being enforced/checked yet (so it will certainly change). Kudos to other developers for figuring this out so quickly...
> 
> `https://api.locastnet.org/api/user/login?client_id=CqhAMsBw%2BnxTXSJMLGqyOw%3D%3D`
> 
> credentials = {
> "username": username,
> "password": password,
> "captcha": "anything"
> }

This worked for me.  I altered line 55, of the file, lib/locast_service.py to match the above (see below code snippet)

`loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login?client_id=CqhAMsBw%2BnxTXSJMLGqyOw%3D%3D',
                                          ('{"username":"' + username + '","password":"' + password + '", "captcha": "anything"}').encode("utf-8"),
                                          {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})`


Thank you jg-24.

**Comment by tmwebb2** [comment]
This worked for me as well. Thank you all for jumping on this so quickly.

**Comment by seakurt** [comment]
Worked here too! Thank you soooooo much!

**Comment by SVRapture** [comment]
I hate to be so stupid (I know I am missing something). I am running locast2plex on a Windows10 box. I made the change above to the locast_service.py file but it still would not work. So I hard coded my username in the locast.service.py file and it is still using the one in my config.ini file. Clearly it is not using the changed .py file. Do I need to compile it or something?

**Comment by jwarner75** [comment]
I am having this very same issue.

I have 2 machines, 1 a Quad Core running Locast2plex and 1 8 Core running Locast2plex both connected to the network and I just had this issue. Both of them are not wanting to connect.

I even made the changes to the file as suggested above and it's not helping either.

Some help would be appreciated to help me get up and running!

Thanks!

**Comment by mazinkhu** [comment]
Can someone please advice where do you find the file locast_service.py, when using locast2plex as docker container  in Synology NAS.

**Comment by Nbr1Sniper** [comment]
> > For the developers: try using a hardcoded client ID and captcha in the locast api calls. The ID is currently hardcoded in website static objects so it may change and the captcha appears have just been added as a required field but is not being enforced/checked yet (so it will certainly change). Kudos to other developers for figuring this out so quickly...
> > `https://api.locastnet.org/api/user/login?client_id=CqhAMsBw%2BnxTXSJMLGqyOw%3D%3D`
> > credentials = {
> > "username": username,
> > "password": password,
> > "captcha": "anything"
> > }
> 
> This worked for me. I altered line 55, of the file, lib/locast_service.py to match the above (see below code snippet)
> 
> `loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login?client_id=CqhAMsBw%2BnxTXSJMLGqyOw%3D%3D', ('{"username":"' + username + '","password":"' + password + '", "captcha": "anything"}').encode("utf-8"), {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})`
> 
> Thank you jg-24.

THANK YOU!!! If anyone can't find the file its in locast2plex\lib and named: locast_service

**Comment by jwarner75** [comment]
> > > For the developers: try using a hardcoded client ID and captcha in the locast api calls. The ID is currently hardcoded in website static objects so it may change and the captcha appears have just been added as a required field but is not being enforced/checked yet (so it will certainly change). Kudos to other developers for figuring this out so quickly...
> > > `https://api.locastnet.org/api/user/login?client_id=CqhAMsBw%2BnxTXSJMLGqyOw%3D%3D`
> > > credentials = {
> > > "username": username,
> > > "password": password,
> > > "captcha": "anything"
> > > }
> > 
> > 
> > This worked for me. I altered line 55, of the file, lib/locast_service.py to match the above (see below code snippet)
> > `loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login?client_id=CqhAMsBw%2BnxTXSJMLGqyOw%3D%3D', ('{"username":"' + username + '","password":"' + password + '", "captcha": "anything"}').encode("utf-8"), {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})`
> > Thank you jg-24.
> 
> THANK YOU!!! If anyone can't find the file its in locast2plex\lib and named: locast_service

I found the file. And I made the change just like that.

I'll list it below to show my modification.

 ```
       loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login',
                                          ('{"username":"' + username + '","password":"' + password + '", "captcha": "anything"}').encode("utf-8"),
                                          {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})
```

I just added it in, I didn't copy the code from here and copied it over.

**Comment by jwarner75** [comment]
Ok, Last comment lol.

I'm SOOO Very new to this...

But I "Forked" the file. Made changes to the Locast file as directed above with the changes. Then I uploaded the file with the changes.

So I think all you got to do is download the amended files, copy over existing one (Except for the Config.ini), you'll wanna make a copy of that so you don't over right the info and have to enter it all again. Then copy it over.
![Screenshot 2021-08-04 181819](https://user-images.githubusercontent.com/88461231/128262640-05b8e842-9c1a-43cc-bfe1-2e6c14547fca.png)

Here is a Screen Shot of the changes made from the file.

**Comment by joerich** [comment]
purduepete

Thank you so much! your fix worked perfectly!

**Comment by bulfinch** [comment]
Any advice for us non developer types running this in docker on synology?

**Comment by ratherDashing** [comment]
@bulfinch 
I fixed it in the official docker container by doing this:
`docker cp locast2plex:/app/lib/locast_service.py .`
then edit the file with the code above then copied it back
`docker cp locast_service.py locast2plex:/app/lib/`
then restarted the container.  make sure the indentation matches up with the previous code (which I just commented out) as python cares about that stuff.

**Comment by jaimelalone** [comment]
Thanks all for the fix.  Best to simply copy pasta!

**Comment by LLCoolWas** [comment]
Is this still working for those of you that applied the change? I tried editing in the client_id into my docker container's files, but it and copying the locast_service.py file from @jwarner75 but it is still giving me an invalid creds error since the update.

**Comment by jwarner75** [comment]
@LLCoolWas Wish I could help you with this issue. I don't know anything about Docker Containers.

I WAS going to install Docker if I couldn't get Python to work when I originally started having some issues with it. It just took me getting away from it for a little while.

I wish I knew MORE about python and the programming.

it took me a few minutes to fix the issue i was having with the Client ID that someone else figured out to get it working. it turned out i was missing a line.

Once i figured it out, I just saved it and I forked the codes and saved the change to the file and uploaded it to the forked files that way someone could just download the updated file, copy over top and should be able to get it to work.

I didn't think about the Docker, Someone else would probably be better to assist you on that that is more knowledgeable about it than me.

**Comment by jwarner75** [comment]
@LLCoolWas it looks like @ratherDashing has a fix for it, Check out his method in the post comments

**Comment by mazinkhu** [comment]
Hi,
Thank you everyone for the help. For those of you who are new to all this like me. Just some detailed steps of how I accessed the files. I am using the Synology NAS DS220+, and do not have a software background. 
FYI: I used the container from Synology Docker registry as shown below:



step1: You will need to access a ssh terminal, to use the docker cp commands. 
step 2: You will need to know the name of the container for you installation. This can be done by clicking the "Edit" tab under the docker container.


step 3: In the ssh terminal window, you will issue the docker cp command to copy the file.Please make sure you use the container name from step2 for the "docker cp" command.(you will need to be root for all the commands below)
>docker cp  tgorg-locast2plex1:/app/lib/locast_service.py .
You can then use the ls command as shown below, to see if the file get copied.


 

Step 4: Modify the copied file, as indicated in the forum. 
Step 5: Copy back the file to the docker container.
> cp locast_service.py  tgorg-locast2plex1:/app/lib/locast_service.py 
I hope this helps.    On Thursday, August 5, 2021, 08:31:09 AM PDT, jwarner75 ***@***.***> wrote:  
 
 


@LLCoolWas it looks like @ratherDashing has a fix for it, Check out his method in the post comments

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by mazinkhu** [comment]
There was a problem with the images, so I added the images,
FYI: I used the container from Synology Docker registry as shown below:
![a1](https://user-images.githubusercontent.com/88461594/128395008-bb0f1bd0-14e7-4f8b-9675-5b8a5368f5b9.png)



step1: You will need to access a ssh terminal, to use the docker cp commands. 
step 2: You will need to know the name of the container for you installation. This can be done by clicking the "Edit" tab under the docker container.
![i2](https://user-images.githubusercontent.com/88461594/128394728-67ea9ef5-1ea7-491c-93ee-540b65a8d18c.png)


step 3: In the ssh terminal window, you will issue the docker cp command to copy the file.Please make sure you use the container name from step2 for the "docker cp" command.(you will need to be root for all the commands below)
docker cp  tgorg-locast2plex1:/app/lib/locast_service.py .
You can then use the ls command as shown below, to see if the file get copied.

![i3](https://user-images.githubusercontent.com/88461594/128394877-aeeddb69-744c-4f60-b453-6c6609cce1df.png)



 

Step 4: Modify the copied file, as indicated in the forum. 
Step 5: Copy back the file to the docker container.

> cp locast_service.py  tgorg-locast2plex1:/app/lib/locast_service.py

**Comment by tgorgdotcom** [comment]
Hello all.  Thanks for all your work into figuring this out.  Let me see if I can integrate the changes to a new version.

**Comment by jwarner75** [comment]
@tgorgdotcom I'm no expert on writing scripts, I forked the files and I made changes to the file that was needed and saved it, I wasn't sure if i could upload to your files or not so I just forked, made changes and uploaded it.

**Comment by rkendry** [comment]
I use Unraid and I cannot figure out where the files are stored to modify the locast_service.py. i might need to just wait for an update to the docker image.

**Comment by patrick-GH** [comment]
> I use Unraid and I cannot figure out where the files are stored to modify the locast_service.py. i might need to just wait for an update to the docker image.

You have to pull the file out of the container then add it back in. I'm no expert either, but will take a look tonight

**Comment by drewwats** [comment]
This fix did not work for me. Latest locast2plex image, Docker container on Synology.

**Comment by mazinkhu** [comment]
Can you please post your modifications, as I  have had no problems. Please make sure the indentation does not change for the lines modified. Below are my modifications, I commented out the old lines and matched the original  indentation.
![Capture](https://user-images.githubusercontent.com/88461594/128435734-707c5dec-9bf6-48bb-8af0-cb2e87eab464.PNG)

**Comment by tgorgdotcom** [comment]
Hello all.  I. just pushed 0.6.6 out to GitHub.  It may take a little bit before a docker image is created for it.

**Comment by mark-in-dallas** [comment]
Worked for me too.  Thanks so much guys!

**Comment by drewwats** [comment]
> Hello all. I. just pushed 0.6.6 out to GitHub. It may take a little bit before a docker image is created for it.

Awesome, this worked! Much appreciated!

`        
        #loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login',
        #                                  ('{"username":"' + username + '","password":"' + password + '"}').encode("utf-8"),
        #                                  {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})

        loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login?client_id=CqhAMsBw%2BnxTXSJMLGqyOw%3D%3D',
                                          ('{"username":"' + username + '","password":"' + password + '","captcha":"locast2plex"}').encode("utf-8"),
                                          {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})`

**Comment by patrick-GH** [comment]
how long does it take for docker hub to get updated?

**Comment by tgorgdotcom** [comment]
Hmm.  It seems that automated builds were disabled on my docker account - moved behind the pro plan.  Sending a request to get it reenabled, since it's an open source project.  Meanwhile I'll see if I can manually push an image to docker.

**Comment by Nbr1Sniper** [comment]
Docker vs windows python script; any advantage or disadvantage? I can run docker in my home assistant instance but figure my pc processor would be better to use over a VM of home assistant. Thoughts?

**Comment by mlweber62** [comment]
The solution provided worked for a day, but is no longer working for me, and I'm getting the login error again.

**Comment by seantdavidson** [comment]
I have upgraded to the new docker version and I am still getting the error

`
Logging into Locast using username emailaddress@email.com...
Error in function login: Forbidden
Invalid Locast Login Credentials. Exiting...`

**Comment by drewwats** [comment]
> I have upgraded to the new docker version and I am still getting the error
> 
> ` Logging into Locast using username emailaddress@email.com... Error in function login: Forbidden Invalid Locast Login Credentials. Exiting...`

The Docker image for locast2plex has not been updated yet. You can manually modify the .py file as described above as a temporary fix. 

https://hub.docker.com/r/tgorg/locast2plex

**Comment by LLCoolWas** [comment]
Ok I finally got time to take another look at this and realized what I was doing wrong. I edited the file on my C: drive where locast2plex's files were, but did not copy it to the container. To do so I had to open a command prompt and run the docker command above (and get my container name right...
![image](https://user-images.githubusercontent.com/11539247/128559767-9d86e6a2-27e1-48f2-bc94-c0879ee0fabf.png)
`
docker cp locast_service.py locast2plex-main_locast2plex_1:/app/lib/locast_service.py
`

Thanks everyone

**Comment by tgorgdotcom** [comment]
Hello all.  Docker image is pushed live now.  Let me know if you're still running into any issues.  I've also sent along a request to locast to see if we can find a more official way to login.

**Comment by drewwats** [comment]
> Hello all. Docker image is pushed live now. Let me know if you're still running into any issues. I've also sent along a request to locast to see if we can find a more official way to login.

Image pulled and started up without issue. Thank you!

**Comment by seantdavidson** [comment]
> Hello all. Docker image is pushed live now. Let me know if you're still running into any issues. I've also sent along a request to locast to see if we can find a more official way to login.

Thanks for all involved, just pulled the newest docker container and I am back up and running.

**Comment by ljo123** [comment]
Still getting the same issue today after upgrading to the newest version of locast2plex. I checked locast_service.py and it has the "captcha":"locast2plex" in there correctly. Not sure what the issue is.

```
Creating locast2plex-066_locast2plex_1 ... done
Attaching to locast2plex-066_locast2plex_1
locast2plex_1  | Locast2Plex v0.5.3
locast2plex_1  | Tuner count set to 3
locast2plex_1  | UUID set to: xxx...
locast2plex_1  | Logging into Locast using username xxx@xxx.com...
locast2plex_1  | Error in function login: Forbidden
locast2plex_1  | Exiting...
```

Never mind. It was stuck using an old image. Working now thanks!

**Comment by zatmonkey** [comment]
Still having the issue today :(
Pulled in a clean image so that's not that and it was working before 
![image](https://user-images.githubusercontent.com/485530/129279797-649fe5f0-bc1c-4d0a-99ef-96aaa47c52de.png)

**Comment by KBoz** [comment]
I tried rebuilding the docker with the latest version 0.6.6 using the docker run -v with tgorg/locast2plex at the end.  For some reason it picked up 0.6.5 instead of the latest on the GitHub.  Version 0.6.5 immediately failed on the login.

Also I'm not sure what version I ws currently runnign anyway.  The only files I can find on my QNAP is a locast2plex-0.5.3 under home on DataVol1

![Screenshot (12)](https://user-images.githubusercontent.com/47510547/129425615-964fa3d0-c6ef-4ca4-9d3e-470e36ced1f1.png)

**Comment by LSparks608** [comment]
Same, cant connect with new version v0.6.6

Error in function login: Not Found
Invalid Locast Login Credentials. Exiting...

**Comment by enz1ey** [comment]
I am also unable to log in, I'm getting this:

Error in function login: Expecting value: line 1 column 1 (char 0)

**Comment by craigarno** [comment]
Seattle WA area, problem described here started Friday 20-Aug-2021
Added code from the login line above (both "clientID" and "captcha" parts) and Locast2Plex-0.6.5 is working again.
Thank you all for finding a fix.

**Comment by scrawfor** [comment]
The docker image does seem stale.  I built it locally and it works fine

```
services:
  locast2plex:
    image: tgorg/locast2plex
    build:
      context: ./
      # dockerfile: ./Dockerfile
    ports:
      - "6077:6077"
      - "1900:1900/udp"
    restart: unless-stopped
    volumes:
      - ./config.ini:/app/config/config.ini

```

**Comment by mikedizzle** [comment]
Thanks for all who commented in this issue!

**Comment by cklinker** [comment]
Register the app at https://www.locast.org/developers/apply to get a valid client id and the captcha issues will go away

**Comment by ghost** [comment]
I am still getting this error after upgrading:

```
Invalid Locast Login Credentials. Exiting...
```

Is anyone else?

**Comment by SuperstationWTF** [comment]
sillydino > Yes,  I am getting the same error as you. It was working yesterday but not today. reinstalled, same error. Changes PW yesterday to alpha/numbers.

**Comment by wouterdebie** [comment]
It seems locast.org implemented the captcha, which breaks login.

**Comment by ghost** [comment]
It sounds like @cklinker 's suggestion is the only way locast2plex will be able to work consistently then?

**Comment by mikedizzle** [comment]
Is locast going to give dev access to all the people who are just running locast2plex on their various machines? 

cc: @cklinker

**Comment by cklinker** [comment]
Modify the code to use this client_id 9qXBrVzpTjUZmVGsZRnnWQ-7GvGeJ48QWtV9v+bsen4=

        loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login?client_id=9qXBrVzpTjUZmVGsZRnnWQ-7GvGeJ48QWtV9v%2Bbsen4%3D',
                                          ('{"username":"' + username + '","password":"' + password + '"}').encode("utf-8"),
                                          {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})

**Comment by mikedizzle** [comment]
@cklinker That worked. Thanks! I did fill out the form to see if I can get my own client ID.

**Comment by patrick-GH** [comment]
@mikedizzle, let us know how that goes

**Comment by tgorgdotcom** [comment]
Hmmm I think I can request for an id and use it for all future versions.  Let me see if I can get an id

**Comment by wouterdebie** [comment]
> Modify the code to use this client_id 9qXBrVzpTjUZmVGsZRnnWQ-7GvGeJ48QWtV9v+bsen4=

Would you mind sharing how you got that ID? I tried figuring out how the android app sent it, but had no luck sniffing traffic with a proxy. I didn't get lucky with trying to get it from decompiling some sources.

**Comment by tgorgdotcom** [comment]
Hello @wouterdebie, you can get a new client ID by registering on https://www.locast.org/developers/apply.  I'm working on getting one myself.  If it's all right with everyone, I'll use the one @cklinker made for now...

The downside to registering is the potential of having to remove location override features from our scripts/apps (per the dev agreement).  I think Locast support has been pretty good dealing with users in the past when they ask them to fix geolocation issues, so the workaround may be to redirect users to Locast when the detected location is wrong.

**Comment by tgorgdotcom** [comment]
Hello all.  0.6.7 includes @cklinker's client ID temporarily.  This is live on GitHub and in docker

**Comment by craigarno** [comment]
> Hello all. 0.6.7 includes @cklinker's client ID temporarily. This is live on GitHub and in docker

Yeah, it wasn't clear if each instance of Locast2Plex had to have its own ClientID, or if the ClientID belonged to the Locast2Plex Application.  So far I'm assuming the later and also updated my running instance to use @cklinker 's provided ClientID (Thank you).  If this is wrong, I suspect we'll hear about it.

My thinking is it wouldn't make sense that every user would have to sign up as a developer to obtain a ClientID.  It makes much more sense that a single ClientID would be for deployment of the Developer's Application, in this case, Locast2Plex.

**Comment by cklinker** [comment]
> > Modify the code to use this client_id 9qXBrVzpTjUZmVGsZRnnWQ-7GvGeJ48QWtV9v+bsen4=
> 
> Would you mind sharing how you got that ID? I tried figuring out how the android app sent it, but had no luck sniffing traffic with a proxy. I didn't get lucky with trying to get it from decompiling some sources.

I created the Locast API.  I have approved the requests for a client id as well and provided the client id I sent in this channel for all requests as well.

**Comment by jwarner75** [comment]
> Hello @wouterdebie, you can get a new client ID by registering on https://www.locast.org/developers/apply. I'm working on getting one myself. If it's all right with everyone, I'll use the one @cklinker made for now...
> 
> The downside to registering is the potential of having to remove location override features from our scripts/apps (per the dev agreement). I think Locast support has been pretty good dealing with users in the past when they ask them to fix geolocation issues, so the workaround may be to redirect users to Locast when the detected location is wrong.

I just applied for a Client ID, Any Idea how long it takes to get approved?

For now I'm using the one that a member created just to keep my Recordings going I have set. Once mine comes in I'll edit the file and add my personal one in.

**Comment by mikesul76** [comment]
Many thanks to all who keep this working with the changes being made by Locast regarding login.  Having a 'software based' DVR tuner that gives me whole house access and remote access (plex server is on 24/7 in my Locast location) is totally awesome!  Who the heck wants to buy yet another piece of hardware for a DVR when we have this.  THANK YOU!

**Comment by Ned-Racine** [comment]
Ditto to mikesul76! Much appreciations to all....

**Comment by mikedizzle** [comment]
Yes @mikesul76! This is awesome for those of us who don't have the chops to pull of emulating this function!

**Comment by jwarner75** [comment]
I applied for a Client_ID so I'd have one myself. I finally got it day before yesterday i believe it was.

Not sure if I did it wrong or what, But when I changed the Information where it needed to be changed. I ended up with an error, it wouldn't work.

Any Idea why that might be?

**Comment by jwarner75** [comment]
> Modify the code to use this client_id 9qXBrVzpTjUZmVGsZRnnWQ-7GvGeJ48QWtV9v+bsen4=
> 
> ```
>     loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login?client_id=9qXBrVzpTjUZmVGsZRnnWQ-7GvGeJ48QWtV9v%2Bbsen4%3D',
>                                       ('{"username":"' + username + '","password":"' + password + '"}').encode("utf-8"),
>                                       {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})
> ```

@cklinker This is the very SAME Client_ID that was given to me when I signed up for one.

I tried this ID and my Script would NOT Run, kept getting failed log in attempts, so had to go back to the other one. Well the top one ending in +bsen4=. Not sure why it wasn't working.

**Comment by cklinker** [comment]
> > Modify the code to use this client_id 9qXBrVzpTjUZmVGsZRnnWQ-7GvGeJ48QWtV9v+bsen4=
> > ```
> >     loginReq = urllib.request.Request('https://api.locastnet.org/api/user/login?client_id=9qXBrVzpTjUZmVGsZRnnWQ-7GvGeJ48QWtV9v%2Bbsen4%3D',
> >                                       ('{"username":"' + username + '","password":"' + password + '"}').encode("utf-8"),
> >                                       {'Content-Type': 'application/json', 'User-agent': self.DEFAULT_USER_AGENT})
> > ```
> 
> @cklinker This is the very SAME Client_ID that was given to me when I signed up for one.
> 
> I tried this ID and my Script would NOT Run, kept getting failed log in attempts, so had to go back to the other one. Well the top one ending in +bsen4=. Not sure why it wasn't working.

The actual client_id you were provided is not url Encoded to be sent in the request.  When I approved all these requests for locast2plex I provided the same client id for all users.

**Comment by jwarner75** [comment]
@cklinker Oh ok. I had to go back to using the one you provided in the code. I don't know anything about th eprogramming, I just follow the instructions and edit the files, copy and paste with Notepad++, at least that's what I use for my Raspberry Pi when editing some files.

**Comment by mikesul76** [comment]
A 93-year old judge made a ruling in a New York court today... 
Honest to goodness, you have got to be kidding me, a 93-year old !! 

He ruled "**Error in function login: Forbidden**"   Very sad, very very sad.

**Comment by Nbr1Sniper** [comment]
Hopefully the ruling won’t effect those who already have access to Locast… It will however hurt expanding the service. They will need to apply for loans and hope they can get enough people to donate to pay it back in order to expand the service.

**Comment by chriselgee** [comment]
Well, looks like I'm not alone in getting this  )-:

```
Error in function login: Forbidden
Invalid Locast Login Credentials. Exiting...
```

**Comment by mikesul76** [comment]
@chriselgee   Ensure you have the latest version (v0.6.7) for this to work.

**Comment by chriselgee** [comment]
Ah - that was it!  Thanks @mikesul76 .  (-:  `docker pull...`

**Comment by Nbr1Sniper** [comment]
Guess it doesn’t matter. Locast has shutdown

**Comment by jaimelalone** [comment]
> Guess it doesn’t matter. Locast has shutdown

Yeah, I just got that email as well. Was going to link to a press release but the site hadn't been updated yet.

**Comment by Nbr1Sniper** [comment]
I got one yesterday saying no more donation requests then this morning saying immediately suspending all services. Nothing works and their site says 0 markets. Ugh!!!!!!! I used locast as a backup to my Tablo as my antenna goes out during lightning storms.

**Comment by ratherDashing** [comment]
> I got one yesterday saying no more donation requests then this morning saying immediately suspending all services. Nothing works and their site says 0 markets. Ugh!!!!!!! I used locast as a backup to my Tablo as my antenna goes out during lightning storms.

darn!

**Comment by ghost** [comment]
I wish they would explain a bit more here. Should we shut our donations down? Are they dead-dead now?

**Comment by ratherDashing** [comment]
> I wish they would explain a bit more here. Should we shut our donations down? Are they dead-dead now?

i turned off my donations, they stopped providing their service, i stopped donating

**Comment by Nbr1Sniper** [comment]
> > I wish they would explain a bit more here. Should we shut our donations down? Are they dead-dead now?
> 
> i turned off my donations, they stopped providing their service, i stopped donating

Don’t blame you. I was about to start donating as I had just started using them. It’s all very confusing. 
Now looking at just getting another tuner that I can use with plex.

**Comment by patrick-GH** [comment]
I was gonna increase my donation amount, because it still would be 100%
more cheaper than regular cable.  I'm willing to bet no one would choose to
donate now.

Maybe instead of interrupting the programming, they float something over
the screen for a minute every 15 minutes.

**Comment by Nbr1Sniper** [comment]
More info here https://youtu.be/SI0lF3fJIXY

**Comment by ghost** [comment]
Curious, what are other people thinking of falling back to? I've used Playon to record off the internet, but this doesn't help with local stations....
